packages/apps/src/overlays/Attest.tsx

```diff
@@ -18,13 +19,28 @@ interface Props {
 function Attest ({ className }: Props): React.ReactElement<Props> | null {
   const { t } = useTranslation();
   const { allAccounts } = useAccounts();
-  const { api } = useApi();
+  const { api, isApiReady } = useApi();
+  const [isOldClaimProcess, setIsOldClaimProcess] = useState(true);
+  const [needAttest, setNeedAttest] = useState<Option<EthereumAddress>[]>([]);
+
+  useEffect(() => {
+    if (isApiReady && typeof api.query?.claims?.preclaims !== 'undefined') {
+      setIsOldClaimProcess(false);
+    }
+  }, [api, isApiReady]);
 
   // Find accounts that need attest. They are accounts that
   // - already preclaimed,
   // - didn't sign the attest yet.
   // `claims.preclaims` returns Some() for these accounts.
-  const needAttest = useCall<[Option<EthereumAddress>]>(api.query.claims?.preclaims.multi, [allAccounts])?.filter((opt) => opt.isSome);
+  useEffect(() => {
+    if (!isOldClaimProcess) {
+      api.query.claims.preclaims.multi<Option<EthereumAddress>>(allAccounts)
```

Comment:
using `api` directly and not `useCall` because it needs to be conditionned.

Reply:
Actually, would it be enough to just replace this line with (I added some `?`):

```ts
const needAttest = useCall<[Option<EthereumAddress>]>(api.query?.claims?.preclaims?.multi, [allAccounts])?.filter((opt) => opt.isSome);
```

if api is not ready, or the chain doesn't have the claims pallet, or the claims module doesn't have preclaims, then `needAttest` would be just undefined. Which won't show the banner.

Reply:
yes, that works indeed!